vaa-core/README.md

```diff
@@ -0,0 +1,13 @@
+# `vaa-core`: Core types and utilities shared by the `vaa` modules
+
+The module contains types and constants that all the `vaa` modules (`vaa-data`, `vaa-filters` and `vaa-matching`) handling abstract logic use and which must be aligned for module interoperation.
```

Comment:
As the doc says, this module only contains the abstract types etc. that these likewise abstract modules depend on. The `vaa-app-shared` is specific to the frontend and backend app implementation, whilst the other modules can in theory be used by any implementation with this `vaa-core` as their only dependency.